features/netsocket/cellular/generic_modem_driver/TESTS/unit_tests/default/main.cpp

```diff
@@ -0,0 +1,541 @@
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+//Add your driver's header file here
+#include "OnboardCellularInterface.h"
+#include "UDPSocket.h"
+#include "TCPSocket.h"
+#include "common_functions.h"
+#include "mbed_trace.h"
+#define TRACE_GROUP "TEST"
+
+using namespace utest::v1;
+
+/** How to run port verification tests
+ *
+ *  i)   Copy this file in your implementation directory
+ *       e.g., netsocket/cellular/YOUR_IMPLEMENTATION/TESTS/unit_tests/default/
+ *  ii)  Rename OnboardCellularInterface everywhere in this file with your Class
+ *  iii) Make an empty test application with the fork of mbed-os where your implementation resides
+ *  iv)  Create a json file in the root directory of your application and copy the contents of
+ *       template_mbed_app.txt into it
+ *  v)   Now from the root of your application, enter this command:
+ *       mbed test --compile-list
+ *       Look for the name of of your test suite matching to the directory path
+ *  vi)  Run tests with the command:
+ *       mbed test -n YOUR_TEST_SUITE_NAME
+ *
+ *  For more information on mbed-greentea testing suite, please visit:
+ *  https://docs.mbed.com/docs/mbed-os-handbook/en/latest/advanced/greentea/
+ */
+
+// The credentials of the SIM in the board.
+#ifndef MBED_CONF_APP_DEFAULT_PIN
```

Comment:
As jimmy noted, my objection was that if the default config is inhte header file, the code file stays clean and only implementation there.. scrolling down through 100x lines of ifdef config values is not user friendly . This header file could come with the test in the same folder (default/main.cpp default/default_config.h)